frontend/src/app/spot/components/switch/switch.component.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  forwardRef,
+  HostBinding,
+  Input,
+  Output,
+  ViewChild,
+} from '@angular/core';
+import {
+  ControlValueAccessor,
+  NG_VALUE_ACCESSOR,
+} from '@angular/forms';
+import { populateInputsFromDataset } from 'core-app/shared/components/dataset-inputs';
+
+export type SpotSwitchState = boolean;
```

Comment:
Why add a new type for boolean? I don't expect the type to change, so I'd propose to keep it at boolean

Reply:
In a sense a lazy copy/paste from spot-checkbox, but I thought I actually kinda liked having the interfacing value type represented near the top of the file.

Reply:
:+1:  fine for me if we keep this practice. I just found it weird to see a custom type mapped to a boolean.